src/api/convert-html.js

```diff
@@ -0,0 +1,77 @@
+const { Collection, Item, Request, Response } = require('postman-collection');
+const PostmanHTMLExtraReporter = require('newman-reporter-htmlextra');
+const { logger } = require('../utils/logger');
+
+function generateHTMLReport(summary) {
+    try{
+        logger.info(
+            `Starting the conversion of JSON summary to HTML report for collection '${summary.collection.info.name}'.`
+        );
+
+        var collection = new Collection(summary.collection.toString());
+        collection.name = summary.collection.info.name;
+      
+        summary.run.executions.forEach((exec) => {
+          exec.item = new Item(exec.item);
+          exec.item.setParent(collection);
+          exec.request = new Request(exec.request);
+          exec.response = new Response(exec.response);
+        });
+      
+        summary.collection = collection;
+      
+        //newman uses an event emitter to communicate with the reporters
+        var newmanEventEmitterMock = {
+          on(event, action) {
+            //the 'beforeDone' event is used by the reporters to build their report and add it to the 'exports' list.
+            if (event == 'beforeDone') {
+              this.action = action;
+              this.action();
+            }
+          },
+          summary: summary,
+          exports: [],
+        };
+      
+        PostmanHTMLExtraReporter(newmanEventEmitterMock, {}, {});
+      
+        var htmlReport = newmanEventEmitterMock.exports[0].content;
+      
+        return htmlReport;
+    }
+    catch(error){
+        logger.error(
+            `An error occured while converting JSON summary to HTML report.`,
+            error
+          );
+    }
+}
+
+function validateSummary(req, res) {
```

Comment:
I would prefer to leave the "req" and "res" out of this class and function.

validateSummary should not have the responsibility of filling the HTTP response, but only to returning if a summary if valid or not.

For example you could have `validateSummary(summary) ` which return a string (the validation error) or null (if no error). This would be used in app.js to validate.

Reply:
I've removed this and used input validation instead.